catalog/dags/maintenance/add_license_url.py

```diff
@@ -1,239 +1,235 @@
+    """
+    Add license_url to meta_data batching all records with the same license.
+
+    :param license_group: tuple of license and version
+    :param batch_size: number of records to update in one update statement
+    :param dag_task: automatically passed by Airflow, used to set the execution timeout.
+    """
+    license_, version = license_group
+    license_info = get_license_info_from_license_pair(license_, version)
+    if license_info is None:
+        raise AirflowSkipException(
+            f"No license pair ({license_}, {version}) in the license map."
         )
-        updated_count: int = postgres.run(
-            update_query, autocommit=True, handler=RETURN_ROW_COUNT
+    *_, license_url = license_info
+
+    logging.info(
+        f"Will add `license_url` in `meta_data` for records with license "
+        f"{license_} {version} to {license_url}."
+    )
+    license_url_dict = {"license_url": license_url}
+
+    # Merge existing metadata with the new license_url
+    update_query = dedent(
+        f"""
+        UPDATE image
+        SET meta_data = ({Json(license_url_dict)}::jsonb || meta_data), updated_on = now()
+        WHERE identifier IN (
+            SELECT identifier
+            FROM image
+            WHERE license = '{license_}' AND license_version = '{version}'
+                AND meta_data->>'license_url' IS NULL
+            LIMIT {batch_size}
+            FOR UPDATE SKIP LOCKED
+        );
+        """
+    )
+    total_updated = 0
+    updated_count = 1
+    while updated_count:
+        updated_count = run_sql(
+            update_query,
+            log_sql=total_updated == 0,
+            method="run",
+            handler=RETURN_ROW_COUNT,
+            autocommit=True,
+            dag_task=dag_task,
         )
-        logger.info(f"{updated_count} records updated with {license_url}.")
-        if updated_count:
-            updated_by_license[license_url] = updated_count
         total_updated += updated_count
-    logger.info(f"Updated {total_updated} rows")
-    # Save the invalid_items to S3 as a TSV
-    save_to_s3(aws_conn_id, invalid_items, s3_bucket)
-    return updated_by_license
+    logger.info(f"Updated {total_updated} rows with {license_url}.")
+
+    return total_updated
 
 
-def save_to_s3(aws_conn_id, invalid_items, s3_bucket):
+@task(trigger_rule=TriggerRule.ALL_DONE)
+def report_completion(updated, dag_task: AbstractOperator = None):
     """
-    Save the records with invalid license pairs to S3.
-    :param aws_conn_id: AWS connection id
-    :param invalid_items: The list of dictionaries with the invalid items
-    :param s3_bucket: S3 bucket
+    Check for null in `meta_data` and send a message to Slack with the statistics
+    of the DAG run.
+
+    :param updated: total number of records updated
+    :param dag_task: automatically passed by Airflow, used to set the execution timeout.
     """
-    if not invalid_items:
-        return
+    total_updated = sum(updated)
```

Comment:
`updated` can be None.
```suggestion
    total_updated = sum(updated) if updated else 0
```

Reply:
Good observation!